### DISCUSSION TITLE
Capillary tube imbibition with OSCAR

### DISCUSSION THREAD

**Discussion by GuillaumeBroggi** [GeoChemFoam/GeoChemFoam#50] [Q&A] [answered]
Dear GeoChemFoam developers and community,

I am assessing the suitability of the OSCAR time-stepping approach (implemented in interOSFoam) for my application by replicating the spontaneous imbibition of a capillary tube, a benchmark case well documented in the literature.

The tube has a diameter of $5 \mu m$, is initially filled with air, and should be spontaneously imbibed by water under a pressure delta $\Delta P$ of 0. 

First, according to the [reference paper](https://arxiv.org/abs/2105.10576), $\Delta t_{OSCAR}$ and $\Delta t_{PISO}$ should be defined. I assume $\Delta t_{PISO}$ corresponds to the time step defined in _controlDict_, but how do we set  $\Delta t_{OSCAR}$?

Second, are there any guidelines to define the $pc$ boundary conditions? I cannot reproduce the expected results, and I suspect I did not define properly the boundary conditions.

I attached a simple case reproducing what's described above.

[imbibition_mesh-5e-1_water_air.zip](https://github.com/GeoChemFoam/GeoChemFoam/files/15459174/imbibition_mesh-5e-1_water_air.zip)

**Comment by julienmaes** [comment, marked as answer]
Dear Guillaume,

The capillary relaxation time-step is defined in the fvSolution file as deltaTCR

Currently, OSCAR only works if the boundary condition are constant velocity at inlet (zeroGradient pressure) and fixedValue pressure at the outlet. Usually capillary pressure is taken as zeroGradient for all bc, but it's not clear if that is correct, especially when the invading phase reach the boundary conditions.

I don't believe OSCAR is suitable for spontaneous imbibition, unless you're doing something different than assuming fixedValue pressure on each side. OSCAR separates the viscous forces generated by the injection and the capillary forces. In spontaneous imbibiton, the injection is not fixed but generated by the capillary forces, I don't see how that could work but maybe I'm missing something.

Regards,

**Reply by GuillaumeBroggi** [reply]
Dear Julien,

That's clear; thanks for the details.

Regards,
Guillaume
